package/src/ssr/NextSSRApolloClient.tsx

```diff
@@ -124,6 +124,7 @@ export class NextSSRApolloClient<
                 );
               });
 
+              promise.finally(cleanup);
```

Comment:
This should always get executed, but in the old codepath it would only be executed if a subscriber was attached to the Observable.